src/applications/static-pages/facilities/vet-center/NearByVetCenters.jsx

```diff
@@ -0,0 +1,187 @@
+import LoadingIndicator from '@department-of-veterans-affairs/component-library/LoadingIndicator';
+import { apiRequest } from 'platform/utilities/api';
+import VetCenterInfoSection from './components/VetCenterInfoSection';
+import VetCenterImageSection from './components/VetCenterImageSection';
+import { connect, useDispatch } from 'react-redux';
+import { facilitiesVetCenterAutomateNearby } from '../../../facility-locator/utils/featureFlagSelectors';
+import { fetchFacilityStarted, fetchFacilitySuccess } from '../actions';
+import { calculateBoundingBox } from '../../../facility-locator/utils/facilityDistance';
+import { getFeaturesFromAddress } from '../../../facility-locator/actions';
+
+const NEARBY_VET_CENTER_RADIUS_MILES = 80;
+
+const NearByVetCenters = props => {
+  const [fetchedVetCenters, setFetchedVetCenters] = useState([]);
+  const dispatch = useDispatch();
+
+  const fetchVetCenters = query => {
+    dispatch(fetchFacilityStarted());
+    apiRequest(query, {
+      apiVersion: 'v1',
+    }).then(res => {
+      dispatch(fetchFacilitySuccess());
+      setFetchedVetCenters(res.data);
+    });
+  };
+
+  const fetchUnpublishedVetCenters = () => {
+    const notPublishedFacilities = props.vetCenters
+      .map(
+        v => !v.entity?.entityPublished && v.entity?.fieldFacilityLocatorApiId,
+      )
+      .join(',');
+    fetchVetCenters(`/facilities/va?ids=${notPublishedFacilities}`);
+  };
+
+  const fetchNearbyVetCenters = () => {
+    const mainAddress = props.mainVetCenterAddress;
+    const query = `${mainAddress.addressLine1}, ${mainAddress.locality} ${
+      mainAddress.administrativeArea
+    } ${mainAddress.postalCode}`;
+    getFeaturesFromAddress(query, ({ body: { features } }) => {
+      const coordinates = features[0].center; // [longitude,latitude]
+      const boundingBox = calculateBoundingBox(
+        coordinates[1],
+        coordinates[0],
+        NEARBY_VET_CENTER_RADIUS_MILES,
+      );
+      const params = [
+        'type=vet_center',
+        'page=1',
+        'per_page=5',
+        'mobile=false',
+        `radius=${NEARBY_VET_CENTER_RADIUS_MILES}`,
+        `latitude=${coordinates[1]}`,
+        `longitude=${coordinates[0]}`,
+        ...boundingBox.map(c => `bbox[]=${c}`),
+      ].join('&');
+      fetchVetCenters(`/facilities/va/?${params}`);
+    });
+  };
+
+  useEffect(
+    () => {
+      if (!props.togglesLoading) {
+        if (props.automateNearbyVetCenters) {
+          fetchNearbyVetCenters();
+        } else {
+          fetchUnpublishedVetCenters();
+        }
+      }
+    },
+    [props.togglesLoading],
+  );
+
+  if (props.facilitiesLoading) {
+    return <LoadingIndicator message="Loading facilities..." />;
+  }
+
+  // TODO: consider moving to a separate component
+  const renderVetCenter = (vetCenter, mainVetCenterPhone) => {
+    // TODO - fix unique key warning
+    // Tried adding this to the div but it causes rendering to fail for some reason:
+    // key={vetCenter.id}
```

Comment:
Could this be `key={vetCenter.id || vetCenter.fieldFacilityLocatorApiId}` or even `vetCenter.title.replaceAll(/\s/, '-')`? I'm not sure publishedVetCenters are guaranteed ids.

Reply:
Thanks Ryan - that worked! You da man.